src/main/java/com/google/sps/servlets/QueryFactory.java

```diff
@@ -24,14 +31,67 @@ static Statement buildSchemaQuery(String table) {
     return Statement.newBuilder(query.toString()).bind("name").to(table).build();
   }
 
-  // Construct SQL statement of form SELECT <columns list> FROM <table>
-  static Statement constructQueryStatement(List<ColumnSchema> columnSchemas, String table) {
-    String columns = String.join(", ", columnSchemas.stream().map(ColumnSchema::columnName).collect(Collectors.toList()));
+  // Construct SQL statement of form SELECT <columns list> FROM <table> WHERE <conditions>
+  static Statement constructQueryStatement(Statement.Builder builder, List<ColumnSchema> columnSchemas, String table, HttpServletRequest request) {
+    List<String> columnsList = columnSchemas.stream()
+        .map(ColumnSchema::columnName)
+        .collect(Collectors.toList());
+    String columns = String.join(", ", columnsList);
     String query = String.format("SELECT %s FROM %s", columns, table);
+    builder.append(query);
+    getWhereStatement(builder, columnSchemas, table, request);
+
+    return builder.build();
+  }
 
-    return Statement.newBuilder(query).build();
+  static void getWhereStatement(Statement.Builder builder, List<ColumnSchema> columnSchemas, String table, HttpServletRequest request) {
+    int loopCount = 0;
+    for (ColumnSchema colSchema : columnSchemas) {
+      String colName = colSchema.columnName();
+      String filterValue = request.getParameter(table + "-" + colName);
+      loopCount = addWhere(filterValue, loopCount, builder, colName, colSchema);
+
+      // Deal with primary keys
+      String primaryKey = request.getParameter(colName);
+      loopCount = addWhere(primaryKey, loopCount, builder, colName, colSchema);
+    }
+  }
+
+  private static int addWhere(String value, int loopCount, Statement.Builder builder, String colName, ColumnSchema colSchema) {
+    if (value != null && !value.equals("")) {
+      if (loopCount == 0) {
+        builder.append(" WHERE ");
+      } else {
+        builder.append(" AND ");
+      }
+      appendCondition(value, builder, colSchema.schemaType(), colName);
+      loopCount++;
+    }
+    return loopCount;
   }
 
+  //TODO add types other than primitives
+  private static void appendCondition(String filterValue, Statement.Builder builder, String colType, String colName) {
+    String condString = colName + " = @" + colName;
+    switch (colType) {
+      case "STRING": 
+      case "DATE":
```

Comment:
I am curious does DATE typed column can be bind to a string value? It seems in https://googleapis.dev/java/google-cloud-spanner/latest/com/google/cloud/spanner/ValueBinder.html, there is a to(com.google.cloud.Date value) method, which hints that we need to create a Date variable first, like what you do for Timestamp.

Reply:
It seemed to work when I treated it like a String.

Reply:
It seemed like it worked as a String when I tried it, but I converted to Date just to be safe.